src/System.Runtime/tests/System/String.cs

```diff
@@ -1137,42 +1139,54 @@ public static void TestReplaceString()
+    [InlineData(StringComparison.CurrentCulture, "Hello", "Abc", false)]
+    [InlineData(StringComparison.CurrentCulture, "Hello", c_SoftHyphen + "Hel", true)]
+    [InlineData(StringComparison.CurrentCultureIgnoreCase, "Hello", "Hel", true)]
+    [InlineData(StringComparison.CurrentCultureIgnoreCase, "Hello", "Hello", true)]
+    [InlineData(StringComparison.CurrentCultureIgnoreCase, "Hello", "", true)]
+    [InlineData(StringComparison.CurrentCultureIgnoreCase, "Hello", "HEL", true)]
+    [InlineData(StringComparison.CurrentCultureIgnoreCase, "Hello", "Abc", false)]
+    [InlineData(StringComparison.CurrentCultureIgnoreCase, "Hello", c_SoftHyphen + "Hel", true)]
+    [InlineData(StringComparison.Ordinal, "Hello", "H", true)]
+    [InlineData(StringComparison.Ordinal, "Hello", "Hel", true)]
+    [InlineData(StringComparison.Ordinal, "Hello", "Hello", true)]
+    [InlineData(StringComparison.Ordinal, "Hello", "Hello Larger", false)]
+    [InlineData(StringComparison.Ordinal, "Hello", "", true)]
+    [InlineData(StringComparison.Ordinal, "Hello", "HEL", false)]
+    [InlineData(StringComparison.Ordinal, "Hello", "Abc", false)]
+    [InlineData(StringComparison.Ordinal, "Hello", c_SoftHyphen + "Hel", false)]
+    [InlineData(StringComparison.OrdinalIgnoreCase, "Hello", "Hel", true)]
+    [InlineData(StringComparison.OrdinalIgnoreCase, "Hello", "Hello", true)]
+    [InlineData(StringComparison.OrdinalIgnoreCase, "Hello", "Hello Larger", false)]
+    [InlineData(StringComparison.OrdinalIgnoreCase, "Hello", "", true)]
+    [InlineData(StringComparison.OrdinalIgnoreCase, "Hello", "HEL", true)]
+    [InlineData(StringComparison.OrdinalIgnoreCase, "Hello", "Abc", false)]
+    [InlineData(StringComparison.OrdinalIgnoreCase, "Hello", c_SoftHyphen + "Hel", false)]
+    public static void TestStartsWith(StringComparison comparisionType, string text, string value, bool expected)
     {
-        String s = "Hello";
-        bool b;
-        b = s.StartsWith("Hel");
-        Assert.True(b);
-        b = s.StartsWith("Hello");
-        Assert.True(b);
-        b = s.StartsWith("");
-        Assert.True(b);
-        b = s.StartsWith("HELLO");
-        Assert.False(b);
-
-        Assert.Throws<ArgumentNullException>(
-            delegate ()
-            {
-                s.StartsWith(null);
-            });
-
-        b = s.StartsWith("Hel", StringComparison.CurrentCultureIgnoreCase);
-        Assert.True(b);
-        b = s.StartsWith("Hello", StringComparison.CurrentCultureIgnoreCase);
-        Assert.True(b);
-        b = s.StartsWith("", StringComparison.CurrentCultureIgnoreCase);
-        Assert.True(b);
-        b = s.StartsWith("HEL", StringComparison.CurrentCultureIgnoreCase);
-        Assert.True(b);
-        b = s.StartsWith("Goodbye", StringComparison.CurrentCultureIgnoreCase);
-        Assert.False(b);
+        if (comparisionType == StringComparison.CurrentCulture)
+        {
+            Assert.Equal(expected, text.StartsWith(value));
+        }
```

Comment:
I've seen this pattern done (and done it my self) in a few places for code coverage purposes (now you get `String.StartsWith(String)` covered)).

Personally, I could go either way.